### PR TITLE
Update destination-header-provider API

### PR DESCRIPTION
**Pull request by newtork**
## What Has Changed?

Update documentation towards upcoming DestinationHeaderProvider API in `3.38.0`. Expected release 12th of February.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs/java/features/connectivity/destinations.mdx [hunk 1]
```diff
@@ -149,24 +149,26 @@ You can implement custom logic that defines which headers are sent under which c
...
+For each registered implementation it invokes the method `getHeaders(DestinationRequestContext)` and appends the returned headers to the currently prepared request.
 
 ### Implement Own Header Provider
 
 #### Implement the Interface
 
 To implement your custom destination header provider, you need to implement the interface `DestinationHeaderProvider`.
-The method `getHeaders` lets you access the destination properties over which the currently prepared request will be issued.
-You can read destination properties (for example, the name and the URI) if your logic requires to determine the headers.
+The method `getHeaders` provides access to the request specific context. The method argument of type `DestinationRequestContext` lets you access the general properties over which the currently prepared request will be issued.
```
Comment (artemkovalyov) [inline docs/java/features/connectivity/destinations.mdx:161]:
```suggestion
The method `getHeaders` provides access to the request specific context.
The method argument of type `DestinationRequestContext` lets you access the general properties over which the currently prepared request will be issued.
```

**Review by artemkovalyov — CHANGES_REQUESTED** [review]
@newtork , this should make pipeline work

**Comment by artemkovalyov** [comment]
@newtork, you can always check linter errors here: 
image
They are much easier to see.

**Review by artemkovalyov — APPROVED** [review]
@newtork, this is nice and good to go, thanks!

**Comment by newtork** [comment]
@artemkovalyov Thanks for taking care :)

**Pull request merged**